state/state.go

```diff
@@ -379,6 +379,7 @@ func (s *State) UpgradeChannel(b *ui.Task, pkg *manifest.Package) error {
 		b.Warnf("Could not check updates for %s. Skipping update. Error: %s", name, err)
```

Comment:
Don't we need it here too? Or perhaps just add that `if etag == "" { etag = pkg.ETag }` below?